src/sentry/services/hybrid_cloud/user.py

```diff
@@ -0,0 +1,135 @@
+from abc import abstractmethod
+from dataclasses import dataclass
+from inspect import signature
+from typing import Any, Iterable, List, Optional, Sequence, Union
+
+from sentry.models.group import Group
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.user import User
+from sentry.services.hybrid_cloud import (
+    CreateStubFromBase,
+    InterfaceWithLifecycle,
+    silo_mode_delegation,
+)
+from sentry.silo import SiloMode
+
+
+@dataclass(frozen=True)
+class APIUser:
+    id: int = -1
+    name: str = ""
+    email: str = ""
+    username: str = ""
+    actor_id: int = -1
+
+    # TODO: Extract to base dataclass?
+    @classmethod
+    def from_orm(cls, orm: Any):
+        # fetch the constructor's signature
+        cls_fields = {field for field in signature(cls).parameters}
+
+        # isolate params we care about
+        args = {}
+        for name, val in orm.__dict__.items():
+            if name in cls_fields:
+                args[name] = val
+
+        ret = cls(**args)
+        return ret
+
+
+class UserService(InterfaceWithLifecycle):
+    @abstractmethod
+    def get_many_by_email(self, email: str, is_active=True) -> List[APIUser]:
+        """
```

Comment:
This should actually be renamed and clarified.  Based on its usage, it looks like the point of this is to fetch *case insensitive email matches* specifically.  There should only be one user per email, but there can be multiple users per case insensitive emails.

Reply:
There wasn't actually any usage of case sensitive fetching of emails, so I removed the bifurcation to reflect actual usage

Reply:
Oh snap, well cool.  👍   More clarity is good.